grails-app/views/species/_names.gsp

```diff
@@ -146,7 +146,80 @@
+            <g:each in="${tc.pullCommonNames}" var="cn">
+                <g:set var="name" value="${cn.nameString}"/>
+                <g:set var="language" value="${cn.language}"/>
+                <g:set var="infoSource" value="${cn.infoSourceURL}"/>
+                <g:set var="isInfoSourceURL" value="${infoSource && infoSource.matches('[a-z]+://.*')}"/>
+                <g:set var="datasetURL" value="${cn.datasetURL}"/>
+                <tr>
+                    <td>
+                        <bie:markLanguage text="${name}" lang="${language}" mark="${!grailsApplication.config.vernacularName.pull.showLanguage}" href="${isInfoSourceURL && infoSource != datasetURL ? infoSource : null}"/>
+                        <bie:markCommonStatus status="common" tags="${cn.labels}"/>
+                        <g:if test="${cn.temporal}"><span title="${message(code: 'label.temporal.vernacular.detail', default: 'Used with a specific time period')}" class="annotation annotation-temporal">${cn.temporal}</span></g:if>
+                        <g:if test="${cn.locality || cn.countryCode}"><span title="${message(code: 'label.locality.vernacular.detail', default: 'Used in a specific locality or country')}" class="annotation annotation-locality">${cn.locality} <bie:country code="${cn.countryCode}"/></span></g:if>
+                        <g:if test="${cn.sex}"><span title="${message(code: 'label.sex.vernacular.detail', default: 'Used with a specific sex')}" class="annotation annotation-sex">${cn.sex}</span></g:if>
+                        <g:if test="${cn.lifeStage}"><span title="${message(code: 'label.lifeStage.vernacular.detail', default: 'Used with a specific life-stage')}" class="annotation annotation-life-stage">${cn.lifeStage}</span></g:if>
+                        <g:if test="${cn.isPlural}"><span title="${message(code: 'label.isPlural.vernacular.detail', default: 'Plural form of name')}" class="annotation annotation-is-plural"><g:message code="label.isPlural.vernacular" default="plural"/></span></g:if>
+                        <g:if test="${cn.organismPart}"><span title="${message(code: 'label.organismPart.vernacular.detail', default: 'Used for a specific part')}" class="annotation annotation-organism-part">${cn.organismPart}</span></g:if>
+                    </td>
+                    <td class="source">
+                        <ul>
+                            <g:if test="${grailsApplication.config.vernacularName.pull.showLanguage}">
+                                <li>
+                                    <bie:showLanguage lang="${language}"/>
+                                </li>
+                            </g:if>
+                            <li>
+                                <g:if test="${cn.datasetURL}"><a href="${cn.datasetURL}"
+                                                                 onclick="window.open(this.href);
+                                                                 return false;">${cn.infoSourceName}</a></g:if>
```

Comment:
onclick attribute can be removed if you add target attribute here.

Reply:
Removed and added